mantidproject/mantid
# Update orsopy to version 1.2.1

### Description of work

#### Summary of work

Updates to the latest version of orsopy, version 1.2.1. This version fixes a bug with column headers in the `.ort` ASCII file, requiring updates to our system test reference files.

For reference, the bug that has been fixed in orsopy is as follows:

The library gives you the flexibility to put multiple datasets in a single file. Each of these datasets can be set up with different numbers of columns, and the column names can also be different for each dataset. However, in orsopy versions up to and including 1.2.0, the saved file continues to use the column headers provided for the first dataset.



Refs #36951. 




### To test:


All tests should pass.

If you would like to check that the files are still output correctly from the Reflectometry GUI then update your Conda environment (`mamba env update -f mantid-developer-YOUROS.yml --prune`) and follow these manual testing instructions (test just the ORSO format output): https://developer.mantidproject.org/Testing/ReflectometryGUI/ReflectometryGUITests.html#save-tab.

**_I have included a release note even though this is just a dependency update because it will fix a user-facing bug in the `.ort` files that we currently output._**





---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
